src/components/Card.js

```diff
@@ -5,17 +5,29 @@ import emoji from 'emoji-dictionary';
 import './Card.css';
 
 class Card extends Component {
+  static propTypes = {
+    id: PropTypes.number.isRequired,
+    text: PropTypes.string,
+    emoji: PropTypes.string,
+    index: PropTypes.number.isRequired,
+    deleteCardCallback: PropTypes.func.isRequired
+  }
+
+  onClickButton = () => {
+    this.props.deleteCardCallback(this);
```

Comment:
I'm not sure if there's a specific rule about this in React, but I think it's generally not considered best practice to pass a component instance itself to a callback.

In the `deleteCard` function from the `Board` component we're only using the `id` and `index` props, so we could just pass those directly instead.